### PR TITLE
Add bbox support

### LINKED ISSUE #7: Support geocoding bbox parameter

### PR DESCRIPTION
**Pull request by mattficke**
Switches to using an options hash instead of another param. Fixes #7.

### CODE REVIEW

**Comment by mattficke** [comment]
@tmcw @sbma44 since you two have contributed the bulk of the code here, can I get 👀 on this? Ok to merge?

**Comment by sbma44** [comment]
I was gonna suggest using Ruby's official URI building library but [apparently it doesn't have one](http://stackoverflow.com/questions/4654708/convert-ruby-hash-to-url-query-string-without-those-square-brackets). Oh well! As long as this passes tests I'm :+1:, though I think you probably could use some of the example code in that link to make it slightly more futureproof for additional arguments passed around in that hash.

File: lib/mapbox/geocoder.rb [hunk 1]
```diff
@@ -6,17 +6,27 @@ class Geocoder
...
-        proximity_param = "?proximity=#{lon}%2C#{lat}"
+        params += "#{params.length > 0 ? '&' : '?'}proximity=#{lon}%2C#{lat}"
       end
 
+      if bbox != nil then
+        minX = bbox[0]
+        minY = bbox[1]
+        maxX = bbox[2]
+        maxY = bbox[3]
+        params += "#{params.length > 0 ? '&' : '?'}bbox=#{minX}%2C#{minY}%2C#{maxX}%2C#{maxY}"
```
Comment (tmcw) [inline lib/mapbox/geocoder.rb]: maybe `bbox=#{bbox.join(',')}"` as a more concise version?

File: lib/mapbox/geocoder.rb [hunk 2]
```diff
@@ -6,17 +6,27 @@ class Geocoder
     include Mapbox::APIOperations::Request
     extend Mapbox::HashUtils
 
-    def self.geocode_forward(query, proximity=nil, dataset='mapbox.places')
-      proximity_param = ''
+    def self.geocode_forward(query, options={}, dataset='mapbox.places')
+      proximity = options[:proximity] || nil
```
Comment (tmcw) [inline lib/mapbox/geocoder.rb]: since the default value of an unset property is `nil`, are these ||s only to save people who put false in their place?

Comment (mattficke) [inline lib/mapbox/geocoder.rb]: Yes, I was trying to keep changes to a minimum and this avoided an undefined method error if `proximity = false`, but I can simplify things and just check `if proximity` instead.

File: lib/mapbox/geocoder.rb (hunk 1 above)
Comment (mattficke) [inline lib/mapbox/geocoder.rb]: 👍

**Comment by mattficke** [comment]
Ok, refactored in response to your comments. I added a handler for extra params we may add in the future that don't need to be reformatted, and tests to illustrate that it works (for both real params like `country` and fake ones).

Since we have tests, I also added this to Circle CI.

Going to go ahead and merge if this looks ok. I think someone else will need to either publish this to rubygems, or give me publish permissions to do so.

**Comment by sbma44** [comment]
LGTM! I don't know who has the keys to rubygems, I'm afraid.

**Comment by mattficke** [comment]
Merged. Looks like @tmcw last published to rubygems.

**Pull request merged**